source/ssb-chart/templates/access-ingress-default-netpol-deny-all.yaml

```diff
@@ -0,0 +1,14 @@
+# Default deny all ingress network traffic
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: "{{ template "app.name" $ }}-ingress-deny-all"
+  labels:
+{{ include "default.labels" $ | indent 4 }}
+spec:
+  podSelector:
+    matchLabels:
+      app: {{ template "app.name" . }}
+  policyTypes:
+    - Ingress
+  ingress: [ ]
```

Comment:
Won't this rule (ingress: []) override any other ingress policy? (Please correct me if I'm wrong).
Perhaps it makes sense to have a rule to block ingress from the ingress-gateway?

Reply:
```
By default, a pod is non-isolated for ingress; all inbound connections are allowed. A pod is isolated for ingress if there is any NetworkPolicy that both selects the pod and has "Ingress" in its policyTypes; we say that such a policy applies to the pod for ingress. When a pod is isolated for ingress, the only allowed connections into the pod are those from the pod's node and those allowed by the ingress list of some NetworkPolicy that applies to the pod for ingress. The effects of those ingress lists combine additively.

Network policies do not conflict; they are additive. If any policy or policies apply to a given pod for a given direction, the connections allowed in that direction from that pod is the union of what the applicable policies allow. Thus, order of evaluation does not affect the policy result.
```
https://kubernetes.io/docs/concepts/services-networking/network-policies/

Reply:
> Network policies are additive, i.e. it is the union set of all network policies that are evaluated. Therefore this policy (block all ingress) will only apply if no other netpol is present.

No, you're right. For some reason I thought it was a distinction between defining an empty "ingress: []" and leaving it out all together, but it's not. So please ignore the comment.